### PR TITLE
Cooperate/GitHub chunqi faultinjection

### PR DESCRIPTION
**Pull request by spring-young**
## What type of PR is this?
/kind feature

## Which issue(s) this PR fixes:
- [x] #12 

### Does this PR introduce a user-facing change?

```release-note

```

### Additional documentation e.g., design docs, usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/proxy/faultinjection/manager.go [hunk 1]
```diff
@@ -0,0 +1,593 @@
...
+		// ValidateRest check
+		logger.Info("start ValidateRest checkrule %s %s", realEndPointUrl.Host, req.Method)
+		validateresult := validator.ValidateRest(req.Header.Get("Mesh-Real-Endpoint"), req.Method)
+		if !validateresult.Allowed {
+			logger.Info("ErrorTProxy: %s %s  ValidateRest NOPASSED ,checkresult:%t, validateresultReason:%s", req.Header.Get("Mesh-Real-Endpoint"), req.Method, validateresult.Allowed, validateresult.Reason)
+			// http.Error(w, fmt.Sprintf("Forbidden by circuit ValidateRest breaker, %s, %s", validateresult.Message, validateresult.Reason), http.StatusForbidden)
+			// return
+		}
+		logger.Info("TProxy: %s %s ValidateRest check PASSED", realEndPointUrl.Host, req.Method)
+
```
Comment (Eikykun) [inline pkg/proxy/faultinjection/manager.go:334]: `withFaultInjection` only takes effect for apiserver requests.
donot validate rest here

File: pkg/apis/ctrlmesh/v1alpha1/faultinjection_types.go [hunk 2]
```diff
@@ -55,41 +57,114 @@ type HTTPFaultInjectionDelay struct {
...
+    Months []int `json:"months,omitempty"`
+}
+
+
+// FaultInjectionState is the status of the fault injection, which may be 'Opened' or 'Closed'.
+type FaultInjectionState string
+
+var (
+	FaultInjectionStatusOpened FaultInjectionState = "Opened"
+	FaultInjectionStatusClosed FaultInjectionState = "Closed"
```
Comment (Eikykun) [inline pkg/apis/ctrlmesh/v1alpha1/faultinjection_types.go:142]: `Opend` and `Closed` generally indicate the fuse state. What does this mean for `FaultInjection`? I don't think `FaultInjection` needs to have a snapshot state.

**Review by Eikykun — COMMENTED** [review]
/ptal

File: pkg/apis/ctrlmesh/v1alpha1/faultinjection_types.go (hunk 2 above)
Comment (spring-young) [reply pkg/apis/ctrlmesh/v1alpha1/faultinjection_types.go:142]: A "Fault Injection snapshot" records whether an error code exception has been triggered; it is a snapshot of the occurrence of the exception.

File: pkg/proxy/faultinjection/manager.go (hunk 1 above)
Comment (spring-young) [reply pkg/proxy/faultinjection/manager.go:334]: add http proxy in main.go

**Comment by spring-young** [comment]
I have read the CLA Document and I hereby sign the CLA

**Review by Eikykun — APPROVED** [review]
/lgtm

**Pull request merged**